ssl-proxy: init at 0.2.7

konst-aa opened the pull request:
## Description of changes
Packaged ssl-proxy.
Homepage: https://github.com/suyashkumar/ssl-proxy
Description taken from the repo: [ssl-proxy is a] simple single-command SSL reverse proxy with autogenerated certificates (LetsEncrypt, self-signed).
<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

eclairevoyant reviewed (changes requested):
please use SRI hash for the fetcher (`hash = "...";` instead of `sha256 = "...";`), and we can also set `meta.mainProgram = "ssl-proxy";` here.

also, any reason to exclude darwin and windows? I see the original repo has support for those.

eclairevoyant reviewed (commented):
also - typo, should be `maintainers` not `maintaners`

konst-aa commented:
> also, any reason to exclude darwin and windows? I see the original repo has support for those.
Yeah, the only issue is that I only have access to a linux machine.

AndersonTorres commented on pkgs/tools/networking/ssl-proxy/default.nix:
```diff
@@ -0,0 +1,33 @@
+
+  vendorHash = "sha256-310K9ZSxy/OQ4HYFCcHQaj4NQwzATrOZ2YkhiSkhY5I=";
+
+  testPhase = ''
```
??

konst-aa replied:
What's the problem?

h7x4 replied:
There is no such thing as a `testPhase`. Maybe you meant `checkPhase`?

https://nixos.org/manual/nixpkgs/stable/#sec-stdenv-phases

konst-aa replied:
Oh, my bad.

nagy commented on pkgs/tools/networking/ssl-proxy/default.nix:
```diff
@@ -0,0 +1,32 @@
+
+  vendorHash = "sha256-310K9ZSxy/OQ4HYFCcHQaj4NQwzATrOZ2YkhiSkhY5I=";
+
+  doCheckPhase = true;
```
I think you mean `doCheck`. But that should be enabled for go modules by default.

konst-aa replied:
Oh yeah, my bad. Programming by coincidence :pensive:.  

I found the line in the `buildGoModule` implementation: `doCheck = args.doCheck or true;` 

Going back to my code though, should I set `doCheck = true;` to be more explicit?

nagy replied:
I dont see other go package do it, so I would say no.

AndersonTorres commented on pkgs/tools/networking/ssl-proxy/default.nix:
```diff
@@ -0,0 +1,31 @@
+    platforms = platforms.linux ++ platforms.darwin ++ platforms.windows ;
+  };
+}
+
```
```suggestion
```

AndersonTorres commented on pkgs/tools/networking/ssl-proxy/default.nix:
```diff
@@ -0,0 +1,31 @@
+{ lib, buildGo118Module, fetchFromGitHub }:
+buildGo118Module rec {
```
```suggestion

buildGo118Module rec {
```

AndersonTorres commented on pkgs/tools/networking/ssl-proxy/default.nix:
```diff
@@ -0,0 +1,31 @@
+{ lib, buildGo118Module, fetchFromGitHub }:
+buildGo118Module rec {
+  version = "0.2.7";
+  pname = "ssl-proxy";
```
```suggestion
  pname = "ssl-proxy";
  version = "0.2.7";
```

konst-aa committed:
ssl-proxy: init at 0.2.7

Ssl-proxy is a reverse proxy that makes it easy to put up and sign ssl
certificates. Homepage: https://github.com/suyashkumar/ssl-proxy

AndersonTorres commented:
Ping me 2 days to merge

konst-aa commented:
@AndersonTorres is it merging time?
